tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationSettings.cs

```diff
@@ -0,0 +1,62 @@
+// <copyright file="RemoteConfigurationSettings.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.RemoteConfigurationManagement
+{
+    internal class RemoteConfigurationSettings
+    {
+        private const int DefaultPollIntervalMilliseconds = 5000;
+
+        public RemoteConfigurationSettings()
+            : this(configurationSource: null)
+        {
+        }
+
+        public RemoteConfigurationSettings(IConfigurationSource configurationSource)
+        {
+            Id = Guid.NewGuid().ToString();
+            RuntimeId = Util.RuntimeId.Get();
+            TracerVersion = TracerConstants.AssemblyVersion;
+            Environment = configurationSource?.GetString(ConfigurationKeys.Environment);
+            AppVersion = configurationSource?.GetString(ConfigurationKeys.ServiceVersion);
```

Comment:
FYI these can also be set on `TracerSettings`

Reply:
I meant that they could _change_ because customers can set `TracerSettings`. We'll address in the "TracerManager" refactoring PR